crates/dbs-virtio-devices/src/balloon.rs

```diff
@@ -343,12 +339,10 @@ impl<AS: DbsGuestAddressSpace, Q: QueueT + Send, R: GuestMemoryRegion>
                     pfn_len
                 );
 
-                let guest_addr = pfn << VIRTIO_BALLOON_PFN_SHIFT;
+                let guest_addr = (pfn as u64) << VIRTIO_BALLOON_PFN_SHIFT;
```

Comment:
Why do you add `as u64` here? To avoid `.into()` below?

Reply:
gpa here should be u64 otherwise it might be overflowed

Reply:
make sense!